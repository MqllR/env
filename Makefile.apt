.PHONY: default
default:
	sudo apt update
	sudo apt install -y \
		ca-certificates \
		cargo \
		curl \
		fish \
		git \
		gnupg \
		lsb-release \
		ripgrep \
		tmux \
		vim
