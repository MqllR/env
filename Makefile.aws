.PHONY: install-cli
install-cli:
	curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
	unzip awscliv2.zip
	sudo ./aws/install
	rm -rf aws awscliv2.zip

.PHONY: install-vault
install-vault:
	curl -L -O "https://github.com/99designs/aws-vault/releases/latest/download/aws-vault-linux-amd64"
	sudo install -o root -g root -m 0755 aws-vault-linux-amd64 /usr/local/bin/aws-vault
	rm -f aws-vault-linux-amd64
