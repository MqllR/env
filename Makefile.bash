# vim:filetype=make

# https://github.com/oh-my-fish/oh-my-fish/
.PHONY: bashrc
bashrc:
	mkdir ${HOME}/.bashrc.d ${HOME}/.bash_completion.d
	curl -L https://raw.githubusercontent.com/bcmarinacci/powerline-shell/master/powerline-shell.bash -o ${HOME}/.bashrc.d/powerline-shell.bash
	curl -L https://raw.githubusercontent.com/cykerway/complete-alias/master/bash_completion.sh -o ${HOME}/.bash_completion/bash_completion.sh
	rm -f ${HOME}/.bashrc
	ln -s ${ROOT_DIR}/dotfiles/.bashrc ${HOME}/.bashrc
