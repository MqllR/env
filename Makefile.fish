# vim:filetype=make

ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))

.PHONY: fish-install
fish-install:
	sudo apt-get install fish
	mkdir -p ${HOME}/.config/fish/functions
	mkdir -p ${HOME}/.config/fish/conf.d
	rm -f ${HOME}/.config/fish/conf.d/mql.fish
	ln -s ${ROOT_DIR}/fish/conf.d/mql.fish ${HOME}/.config/fish/conf.d/mql.fish

# https://github.com/jorgebucaran/fisher
.PHONY: fisher-install
fisher-install: fish-install
	curl https://git.io/fisher --create-dirs -sLo ${HOME}/.config/fish/functions/fisher.fish

# https://github.com/oh-my-fish/oh-my-fish/
.PHONY: omf-install
omf-install: fish-install
	curl -L https://get.oh-my.fish | fish
