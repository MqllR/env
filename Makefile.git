# vim:filetype=make

HUBVERSION=2.14.1
HUBARCHIVE=hub-linux-amd64-${HUBVERSION}
HUBURL=https://github.com/github/hub/releases/download/v${HUBVERSION}/${HUBARCHIVE}.tgz

ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))

.PHONY: git-install
git-install: gitconfig
	sudo apt-get install git

.PHONY: hub-install
hub-install:
	curl -LO ${HUBURL}
	tar xzf ${HUBARCHIVE}.tgz
	sudo ./${HUBARCHIVE}/install
	rm -rf ${HUBARCHIVE}*

.PHONY: gitconfig
gitconfig:
	rm -f ${HOME}/.gitconfig
	ln -s ${ROOT_DIR}/dotfiles/.gitconfig ${HOME}/.gitconfig
