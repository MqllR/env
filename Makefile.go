# vim:filetype=make

GOVERSION=1.14.4
GOARCHIVE=go${GOVERSION}.linux-amd64.tar.gz
GOURL=https://dl.google.com/go/${GOARCHIVE}

ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))

.PHONY: install
install: go-clean
	curl -O ${GOURL}
	sudo tar -C /usr/local -xzf ${GOARCHIVE}
	rm -f ${GOARCHIVE}

.PHONY: go-clean
go-clean:
	sudo rm -rf /usr/local/go

.PHONY: fish
fish:
	rm -f ${HOME}/.config/fish/functions/mql-go.fish
	ln -s ${ROOT_DIR}/fish/functions/mql-go.fish ${HOME}/.config/fish/functions/mql-go.fish

.PHONY: bash
bash:
	rm -f ${HOME}/.bashrc.d/mql-go.sh
	ln -s ${ROOT_DIR}/bashrc/mql-go.sh ${HOME}/.bashrc.d/mql-go.sh
