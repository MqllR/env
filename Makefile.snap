.PHONY: default
default:
	sudo snap install \
		brave

.PHONY: messaging
messaging:
	sudo snap install --classic slack
	sudo snap install \
		zoom-client \
		telegram-desktop
